### DISCUSSION TITLE
Educational module

### DISCUSSION THREAD

**Discussion by fxpar** [sqlitebrowser/sqlitebrowser#2510] [General]
Hi,

Is there a way to add an educational module to allow students to work on exercises about a database:

- check if the results of the query is identical to the expected results
- have a view to compare student's query to the correction (hopefully variants of solutions could be provided)

_(Maybe someone has already done something like that?)_
Sorry if this is not the place for this question.

**Comment by chrisjlocke** [comment]
There isn't such a module, and while 'extensions' can be linked to, this is more SQLite extensions, rather than 'modules' or 'addons' to DB4S itself.
You can't compare results of a query to 'expected results' and I'm not sure how that would work - you couldn't even do something easy like compare the number of rows, as the rows returned could be wrong.  You'd have to compare each column, but even that could be different depending on the select ... and even worse, if you don't specify an 'order by', the order the rows returned could be random.

**Comment by mgrojo** [comment]
To get some ideas, you could check what others are doing to teach SQL using DB4S in https://github.com/sqlitebrowser/sqlitebrowser/wiki/Tutorials
